server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -44,6 +44,7 @@ public enum SettingType {
 
     private Setting<?> setting;
     private SettingType type;
+    private WriteableSetting parser;
```

Comment:
And there it is, we've defined parser as a Setting here.  This should not be here, the parser is part of the `Setting<?>` object.   You need the value of `setting.parser` which is private, so you'll need to create a getter for it.